build-tools/build.sh

```diff
@@ -0,0 +1,217 @@
+#!/bin/bash
+script_start_time=$SECONDS
+# ------------
+# ECHO LICENSE
+# ------------
+echo "::group::License"
+cat /LICENSE
+echo "::endgroup::"
+
+# ----------------
+# SETTING VARIABLES
+# ----------------
+
+set -e # Exit on error
+
+git config --global --add safe.directory /github/workspace
+
+
+# ----------------
+# CHECK IF TEMPLATE
+# ----------------
+echo "::group::Checking if this is a template"
+
+template=$(awk -F'=' '/^IS_LIBRARY:=/{print $2}' Makefile)
+if [ "$template" == "1" ]; then
+    echo "is template"
+else
+    echo "is not template"
+fi
+echo "template=$template" >> $GITHUB_OUTPUT
+
+echo "::endgroup::"
+
+
+# ----------------
+# GET PROJECT INFO
+# ----------------
+echo "::group::Getting project info"
+
+if [ "$GITHUB_EVENT_NAME" == "pull_request" ]; then
+    # Fetch the head SHA directly from the PR API
+    API_URL="https://api.github.com/repos/$GITHUB_REPOSITORY/pulls/$GITHUB_PR_NUM"
+    echo "API URL: $API_URL"
+    API_RESPONSE=$(wget -O- --quiet "$API_URL")
+    if [ $? -ne 0 ]; then
+        echo "Error fetching data from GitHub API"
+        exit $?
+    fi
+    echo "API Response: $API_RESPONSE"
+    sha=$(echo "$API_RESPONSE" | jq -r '.head.sha' | head -c 6)
+
+else
+    # Use the commit SHA after the event
+    sha=$(git rev-parse HEAD | head -c 6)
+fi
+
+
+
+function get_sha() {
+    echo "sha=$sha" >> $GITHUB_OUTPUT
+    echo "SHA found: $sha"
+}
+
+function get_version() {
+    version=$(awk -F'=' '/^VERSION:=/{print $2}' Makefile)
+    echo "Version found: $version"
+    echo "version=$version" >> "$GITHUB_OUTPUT"
+    echo $version >> "version.txt"
+}
+
+function get_library_name() { 
+    library_name=$(awk -F'=' '/^LIBNAME:=/{print $2}' Makefile)
+    echo "library_name=$library_name" >> "$GITHUB_OUTPUT"
+    echo "Library name found: $library_name"
+    echo $library_name >> "library_name.txt"
+}
+
+get_sha &
+get_version &
+get_library_name &
+wait
+
+version=$(cat version.txt)
+rm version.txt
+library_name=$(cat library_name.txt)
+rm library_name.txt
+
+echo "Version before setting postfix: $version"
+echo "SHA before setting postfix: $sha"
+
+if [ "$INPUT_NO_COMMIT_HASH" == "true" ]; then
+    postfix="${version}"
+else
+    postfix="${version}+${sha}"
+fi
+echo "Postfix after setting: $postfix"
+echo "postfix=$postfix" >> "$GITHUB_OUTPUT"
+
+name="$library_name@$postfix"
+echo "name=$name" >> "$GITHUB_OUTPUT"
+echo "Name found: $name"
+
+echo "::endgroup::"
+# ----------------
+# BUILDING PROJECT
+# ----------------
+# Pause errors
+set +e
+pros make clean
```

Comment:
why call pros here? just use `make clean`

Reply:
I was using pros make everywhere, but switched away from that when I decided to make compiling multithreaded with the -j flag.

Reply:
so then use make everywhere then? it also makes it easier to remove pros as a dependency if we decided to do so again